src/CoreWCF.Primitives/src/CoreWCF/Channels/MtomMessageEncoder.cs

```diff
@@ -561,7 +561,14 @@ private async Task WriteMessageInternalAsync(Message message, Stream stream, str
                 await xmlWriter.WriteEndDocumentAsync();
             }
 
-            await xmlWriter.FlushAsync();
+            if (xmlWriter.SupportsAsync())
+            {
+                await xmlWriter.FlushAsync();
```

Comment:
Any reason for not using `ConfigureAwait(false)` whenever using `await` ?

Reply:
I think it's partly explained in https://github.com/CoreWCF/CoreWCF/pull/711